internal/core/services/link.go

```diff
@@ -0,0 +1,26 @@
+package services
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+
+	"github.com/polygonid/sh-id-platform/internal/core/domain"
+	"github.com/polygonid/sh-id-platform/internal/core/ports"
+)
+
+// Link - represents a link in the issuer node
+type Link struct {
+	linkRepository ports.LinkRepository
+}
+
+// NewLinkService - constructor
+func NewLinkService(linkRepository ports.LinkRepository) ports.LinkService {
+	return &Link{linkRepository: linkRepository}
+}
+
+// Save - save a new credential
+func (ls *Link) Save(ctx context.Context, link domain.Link) (*uuid.UUID, error) {
```

Comment:
I think it is better to pass parameters or a DTO request and create the domain.Link entity inside the Save method.  Also, in can be more useful to return the domain.Link entity created and not only the id. So the caller can use the whole entity if needed.

Reply:
in this case I dont see the why is better the dto or the parameters. If we pass a dto, the service must to create the domain link (only that). Same if we pass the parameters. DTOS are useful in other contexts like credential creation. In this context I dont see the benefit. anyway i will change it. Regarding the response, its ok